app/controllers/api/base_controller/renderer.rb

```diff
@@ -206,7 +206,12 @@ def virtual_attribute_search(resource, attribute)
             return resource_attr unless resource_attr.try(:first).kind_of?(ApplicationRecord)
             Rbac.filtered(resource_attr)
           else
-            Rbac.filtered_object(resource).try(:public_send, attribute)
+            obj = Rbac.filtered_object(resource)
+            begin
+              obj.try(:public_send, attribute)
+            rescue => err
```

Comment:
I believe this fix masks model issues in general and should not return the 400 but the 500 it does today.  The request is indeed valid as reconfigure_dialog is a valid relationship for services (OPTIONS /api/services would show that) and thus GET /api/services/:id?attributes=reconfigure_dialog is a valid API request.  The fix should be moved down to the model and in this particular case, the reconfigure_dialog would need to return a nil instead of stack tracing.

Reply:
So I suggested in the description changing 211 to ```Rbac.filtered_object(resource).try(:attribute)``` so that it would, in fact, return nil.

Reply:
AFAIR the try will fail if the attribute/method exists and raises an exception, it will only return nil if it does not exist. The issue here is the begin/rescue masks the error and returns the 400/bad request when it is indeed a good request.

Reply:
I thought reconfigure_dialog was not defined on services...

Reply:
it is as a relationship. attributes= in the API allows one to not only query attributes, virtual attributes, but also relationships, which reconfigure_dialog is seen as such, do an OPTIONS /api/services to see the whole lot.